frontend/src/ArticleEdit/ArticleEdit.js

```diff
@@ -2,81 +2,113 @@ import React from 'react';
 import marked from 'marked';
 import PropTypes from 'prop-types';
 
+import actions from './actions';
+
 import Input from './Input';
 import Tab from './Tab';
-import { Button, Editor, Heading1, Preview, Textarea, Wrapper } from './styled';
+import {
+  Button,
+  Editor,
+  Heading1,
+  Preview,
+  Textarea,
+  Wrapper,
+  Message,
+} from './styled';
 
 const propTypes = {
-  data: PropTypes.shape({
-    topic: PropTypes.string,
-    content: PropTypes.string,
-    sub_topic: PropTypes.array,
-    title: PropTypes.string,
-  }),
+  id: PropTypes.string,
   empty: PropTypes.bool,
 };
 
 const defaultProps = {
-  data: { topic: '', content: '', sub_topic: [], title: '' },
+  id: '0',
   empty: false,
 };
 
 class ArticleEdit extends React.Component {
   state = {
     edit: true,
-    content: this.props.empty ? '' : this.props.data.content,
+    article: {},
+    message: { show: false, error: false, text: '' },
+  };
+
+  componentDidMount = () => {
+    if (this.props.id && this.props.id !== '0') {
+      actions.get(this.props.id).then(article => this.setState({ article }));
+    }
   };
 
   handleChange = e => {
     const { edit } = this.state;
     if (e.target.id === 'radio-preview' && edit) {
-      this.setState(() => ({ edit: false, content: this.textarea.value }));
+      this.setState(() => ({
```

Comment:
Using async setState, you should pass the current state as params, eg:
```
this.setState(({message, article}) => ...
```

Reply:
oh right yes, pasing a function to setState. Blindly editing ^^  Must've updated existing code. I'll read up on the difference, thanx for the catch.
Will change.